src/topology/homeomorph.lean

```diff
@@ -425,3 +425,67 @@ def image (e : α ≃ₜ β) (s : set α) : s ≃ₜ e '' s :=
   ..e.to_equiv.image s, }
 
 end homeomorph
+
+namespace continuous
+variables [topological_space α] [topological_space β]
+
+lemma continuous_symm_of_equiv_compact_to_t2 [compact_space α] [t2_space β]
+  {f : α ≃ β} (hf : continuous f) : continuous f.symm :=
+begin
+  rw continuous_iff_is_closed,
+  intros C hC,
+  have hC' : is_closed (f '' C) := (hC.is_compact.image hf).is_closed,
+  rwa equiv.image_eq_preimage at hC',
+end
+
+/-- Continuous equivalences from a compact space to a T2 space are homeomorphisms.
+
+This is not true when T2 is weakened to T1
+(see `continuous.homeo_of_equiv_compact_to_t2.t1_counterexample`). -/
+@[simps]
+def homeo_of_equiv_compact_to_t2 [compact_space α] [t2_space β]
+  {f : α ≃ β} (hf : continuous f) : α ≃ₜ β :=
+{ continuous_to_fun := hf,
+  continuous_inv_fun := hf.continuous_symm_of_equiv_compact_to_t2,
+  ..f }
+
+/--
+Let `α = ℕ` be the one-point compactification of `{1, 2, ...}` with the discrete topology,
+where `0` is the adjoined point, and let `β = ℕ` be given the cofinite topology.
+Then `α` is compact, `β` is T1, and the identity map `id : α → β` is a continuous equivalence
+that is not a homeomorphism.
+
+This demonstrates that `continuous.homeo_of_equiv_compact_to_t2` cannot be generalized from
+`t2_space` to `t1_space`.
+-/
+lemma homeo_of_equiv_compact_to_t2.t1_counterexample :
+  ∃ (α β : Type) (Iα : topological_space α) (Iβ : topological_space β), by exactI
+  compact_space α ∧ t1_space β ∧ ∃ f : α ≃ β, continuous f ∧ ¬ continuous f.symm :=
+begin
+  let topα : topological_space ℕ := nhds_adjoint 0 filter.cofinite,
```

Comment:
Is it obvious that this is the one-point compactification of the discrete topology on `{1, 2, ...}`? It doesn't matter for the proof of course, but I personally don't see right now why this is the topology that is promised by the docstring.

Reply:
It's obvious after you unfold definitions and think about it for a while :smile:

In the `nhds_adjoint 0 filter.cofinite` topology, a set is open if (1) 0 is not in the set or (2) 0 is in the set and the set is cofinite. That happens to be exactly what the one-point compactification would be.

Reply:
Please include that explanation as a comment.

Reply:
Ok, included